src/sentry/runner/initializer.py

```diff
@@ -60,6 +60,23 @@ def register_plugins(settings, raise_on_plugin_load_failure=False):
             pass
 
 
+def unregister_plugins(settings):
```

Comment:
@wedamija Would like to get your eyes on how I am doing the register/unregister of plugins since this is new.

Reply:
@wedamija I think this PR is in a reviewable state again. Might be worth checking out the actual implementation as well since this route became a bit heavier with my latest changes.

Reply:
I'm wondering if it's worth doing something else here, rather than registering/unregistering plugins at the global level, which I can imagine might fail in various ways down the line.

One option could be to do something like
`mock.patch("sentry.plugins.base.plugins", new=PluginManager())` and then you can just register your changes as usual, and they'll be automatically rolled back once the mock unregisters. You may have to mock `register` and `unregister` too.